scripts/delete_resources.sh

```diff
@@ -1,149 +1,240 @@
+      #Get status from the change
+      aws route53 wait resource-record-sets-changed --id "$STATUS_ID"
+      echo "Successfully deleted Route53 record ${NAME}: status: ${STATUS_ID}"
+      exit
+    done
+    rm -rf "${PWD}"/payload.json
   fi
+}
+
+delete_all_resources () {
+  delete_ec2_instances "$1"
+  delete_db_resources "$1"
+  delete_lb_resources "$1"
+  delete_target_groups "$1"
+  delete_route_s3 "$1"
+}
+
+
+while getopts r:h OPTION
+do 
+  case "${OPTION}"
+    in
+    r) RESOURCES=${OPTARG};;
+    h|?)
+      echo "
+    Usage: 
+      $(basename "$0") [-r <resource_prefix>] [-h]
+
+      -r: resource prefix names. can be comma separated. 
+      -h: help. Prints usage example.
+      
+      Ex 1: 
+      ./delete_jenkins_resources.sh -r resourceprefix1,resourceprefix2,resourceprefix2
+      - This will delete all resources for the names resourceprefix1,resourceprefix2 and resourceprefix2
+      Ex 2:
+      ./delete_jenkins_resources.sh
+      - This will lookup the resource name from local tfvars file and delete the resources. This is interactive and will ask for confirmation before delete
+      Ex 3: 
+      ./delete_jenkins_resources.sh -h
+      - Print usage details for reference.
+      "
+      exit 1
+      ;;
+  esac
+done
+
+if [ "${RESOURCES}" = "" ]; then
+    echo "Working with local .env and .tfvars file to get resource name for deletion"
+    #Get resource name from tfvarslocal && change name to make more sense in this context
+    PRODUCT_NAME=$(grep ENV_PRODUCT <./config/.env | cut -d= -f2 | tr -d ' "')  # Split string based on delimiter =
+    if echo "${PRODUCT_NAME}" | grep -q "ENV_PRODUCT"; then
+      PRODUCT_NAME=$(echo "${PRODUCT_NAME}" | cut -d ":" -f 2)  # Split string based on delimiter :
+    fi
+    if [[ -z "$PRODUCT_NAME" || ! "$PRODUCT_NAME" =~ ^(rke2|k3s)$ ]]; then
+      echo "Wrong or empty product name found in .env file for: $PRODUCT_NAME"
+      exit 1
+    fi
+
+    #Validate path to the tfvars file
+    if [[ ! -f ./config/"$PRODUCT_NAME".tfvars ]]; then
+      echo "No $PRODUCT_NAME.tfvars file found in config directory"
+      exit 1
+    fi
+
+    #Get resource name from tfvars file and validate
+    RESOURCE_NAME=$(grep resource_name <./config/"$PRODUCT_NAME".tfvars | cut -d= -f2 | tr -d ' "')
+    if [[ -z "$RESOURCE_NAME" ]]; then
+      echo "No resource name found for: $PRODUCT_NAME.tfvars file"
+      exit 1
+    fi
+
+    printf "This is going to delete all AWS resources with the prefix %s. Continue (yes/no)? " "$RESOURCE_NAME"
+    read -r REPLY
+    if [[ "$REPLY" =~ ^[Yy][Ee][Ss]$ ]]; then
+      delete_all_resources "${RESOURCE_NAME}"
+    else
+      echo "Exiting: No resources deleted as per user input. Please delete the resources manually"
+      exit 1
+    fi
 else
-  echo "Exiting: No resources deleted as per user input. Please delete the resources manually"
-  exit 1
-fi
\ No newline at end of file
+    for i in $(echo "${RESOURCES}" | tr "," "\n")
```

Comment:
for sake of performance the for loop should be ideally done in the functions.
so you should pass the whole list to delete_all 

and then, any reason why doing this sequentially ? its going to take a huge time running this for a lot of resource names ( since we have that option now) 

we should pass the list to each fucntion, inside the function do a loop with all resource names and on the  caller we should do somthing like 
```
delete_all_resources () {
 delete_ec2_instances "$1" &
  delete_db_resources "$1" &
  delete_lb_resources "$1" &
  delete_target_groups "$1" &
  delete_route_s3 "$1" &

    wait
}
```

Reply:
the deletes happen pretty fast. i hv created like 20 30 resources per group, and the max it has run is maybe 2 mins. the timeout for the job is 30 mins right now. i want it to log the for this prefix - these resources are being deleted - and they should be deleted together in a group. i dont want all ec2 instances all prefix one set, all db clusters all prefix etc. its for easier readability and follow up to  make sure everything got deleted for one prefix first. its log readability that is more important here than performance of a for loop.

Reply:
ok